patches/roslyn-analyzers/0001-WIP-for-source-build.patch

```diff
@@ -0,0 +1,1087 @@
+ RoslynAnalyzers.sln                           |  94 ------
+ build.sh                                      |  16 +
+ eng/GenerateAnalyzerNuspec.targets            |  15 +-
+ .../GenerateAnalyzerNuspec.csproj             |   8 +
+ eng/GenerateAnalyzerNuspec/Program.cs         | 298 ++++++++++++++++++
+ eng/common/dotnet-install.sh                  |   0
+ ...soft.CodeAnalysis.Analyzers.Package.csproj |   4 +-
+ ...soft.CodeAnalysis.AnalyzerUtilities.csproj |   2 +-
+ .../Microsoft.CodeAnalysis.Analyzers.csproj   |   4 +-
+ .../Setup/AnalyzerReleases.Shipped.md         |   0
+ .../Setup/AnalyzerReleases.Unshipped.md       |   0
+ ...oft.CodeAnalysis.BannedApiAnalyzers.csproj |   2 +-
+ .../Setup/AnalyzerReleases.Shipped.md         |   0
+ .../Setup/AnalyzerReleases.Unshipped.md       |   0
+ ...Microsoft.CodeAnalysis.NetAnalyzers.csproj |   2 +-
+ .../Setup/AnalyzerReleases.Shipped.md         |   0
+ .../Setup/AnalyzerReleases.Unshipped.md       |   0
+ ...oft.CodeAnalysis.PublicApiAnalyzers.csproj |   2 +-
+ ...alysis.PublicApiAnalyzers.CodeFixes.csproj |   2 +-
+ .../Setup/AnalyzerReleases.Shipped.md         |   0
+ .../Setup/AnalyzerReleases.Unshipped.md       |   0
+ .../Core/Roslyn.Diagnostics.Analyzers.csproj  |   2 +-
+ .../Setup/AnalyzerReleases.Shipped.md         |   0
+ .../Setup/AnalyzerReleases.Unshipped.md       |   0
+ src/Text.Analyzers/Core/Text.Analyzers.csproj |   2 +-
+ .../Setup/AnalyzerReleases.Shipped.md         |   0
+ .../Setup/AnalyzerReleases.Unshipped.md       |   0
+ 29 files changed, 344 insertions(+), 302 deletions(-)
+ delete mode 100644 FxCopAnalyzers.sln
+ rename nuget.config => NuGet.config (100%)
+ create mode 100755 build.sh
+ create mode 100644 eng/GenerateAnalyzerNuspec/GenerateAnalyzerNuspec.csproj
+ create mode 100644 eng/GenerateAnalyzerNuspec/Program.cs
+ mode change 100644 => 100755 eng/common/dotnet-install.sh
+ create mode 100644 src/Microsoft.CodeAnalysis.Analyzers/Setup/AnalyzerReleases.Shipped.md
+ create mode 100644 src/Microsoft.CodeAnalysis.Analyzers/Setup/AnalyzerReleases.Unshipped.md
+ create mode 100644 src/Microsoft.CodeAnalysis.BannedApiAnalyzers/Setup/AnalyzerReleases.Shipped.md
+ create mode 100644 src/Microsoft.CodeAnalysis.BannedApiAnalyzers/Setup/AnalyzerReleases.Unshipped.md
+ create mode 100644 src/NetAnalyzers/Setup/AnalyzerReleases.Shipped.md
+ create mode 100644 src/NetAnalyzers/Setup/AnalyzerReleases.Unshipped.md
+ create mode 100644 src/PublicApiAnalyzers/Setup/AnalyzerReleases.Shipped.md
+ create mode 100644 src/PublicApiAnalyzers/Setup/AnalyzerReleases.Unshipped.md
+ create mode 100644 src/Roslyn.Diagnostics.Analyzers/Setup/AnalyzerReleases.Shipped.md
+ create mode 100644 src/Roslyn.Diagnostics.Analyzers/Setup/AnalyzerReleases.Unshipped.md
+ create mode 100644 src/Text.Analyzers/Setup/AnalyzerReleases.Shipped.md
+ create mode 100644 src/Text.Analyzers/Setup/AnalyzerReleases.Unshipped.md
+
+diff --git a/FxCopAnalyzers.sln b/FxCopAnalyzers.sln
+deleted file mode 100644
```

Comment:
Is there some other way to selectively disable a `.sln` file?

Reply:
Maybe we can specify one in the build command... but I think we can defer this improvement.